android/src/main/kotlin/cz/dronetag/flutter_opendroneid/FlutterOpendroneidPlugin.kt

```diff
@@ -110,62 +87,137 @@ class FlutterOpendroneidPlugin : FlutterPlugin, ActivityAware, Pigeon.Api {
         )
     }
 
+    override fun initialize(result: Pigeon.Result<Void>) {
+        if(context == null || boundActivity == null) {
+            return result.error(PluginNotAttachedException())
+        }
+        
+        val wifiManager: WifiManager? =
+            context!!.getSystemService(Context.WIFI_SERVICE) as WifiManager?
+        val wifiAwareManager: WifiAwareManager? =
+            context!!.getSystemService(Context.WIFI_AWARE_SERVICE) as WifiAwareManager?
+
+        wifiScanner =
+            WifiScanner(
+                wifiOdidPayloadStreamHandler, wifiStateStreamHandler, wifiManager, context!!
+            )
+        wifiNaNScanner =
+            WifiNaNScanner(
+                wifiOdidPayloadStreamHandler, wifiStateStreamHandler, wifiAwareManager, context!!
+            )
```

Comment:
I think it's already handled since we've not encountered the issue before I think, but what happens if Wi-Fi NAN is not supported? Is it possible? Does it throw an exception and would the plugin be uninitialized then?

Reply:
NAN Scanner handles Wi-FI NAN support itself, it checks whether it is supported when it is created. When NAN is not supported, scanner is normally created, but it reports that NAN is not supported.